init/prepare.sh

```diff
@@ -44,6 +44,8 @@ pushd $SOURCE_DIR/ndt
     export LDFLAGS="-L$BUILD_DIR/build/lib"
     ./bootstrap
     ./configure --enable-fakewww --prefix=$BUILD_DIR/build
+    # Run unit tests on source before making and installing
+    make check || (echo "Unit testing of the source code failed." && exit 1)
```

Comment:
Sorry I missed this before. Why run `make check` before `make`?

The order I expect is `make`, `make check`, `make install`. 

My thinking is that `make` first will identify any compile errors as distinct from unit test errors (from make check). If we run `make check` first, and there are compile errors, the check will fail but the error message will hint that it's a unit test that failed rather than a compile error.

Does this sound right?